server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java

```diff
@@ -77,7 +80,11 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
 
         @Override
         public void enQueue(Mail mail) throws MailQueueException {
-            mailItems.addFirst(new MemoryMailQueueItem(mail));
+            try {
+                mailItems.addFirst(new MemoryMailQueueItem(new MailImpl(mail)));
```

Comment:
Sorry, not sure to understand you answer. As you're duplicating the resource, are you sure there are 2 callings scopes that will dispose it?